nomenclature/region_mapping_models.py

```diff
@@ -5,7 +5,37 @@
 
 import yaml
 from jsonschema import validate, ValidationError
-from pydantic import BaseModel, validator, root_validator
+import pydantic
+from pydantic import BaseModel, validator, root_validator, PydanticValueError
+
+import copy
+
+# We take a deep copy of the original __str__ from
+# pydantic.error_wrappers.ValidationError. We do this to keep the changes minimally
+# invasive and get 'automatic' updates in case of any changes upstream
+original__str__ = copy.deepcopy(pydantic.error_wrappers.ValidationError.__str__)
+
+# Define a new __str__ method which adds file information in case it is present.
+# Otherwise the original __str__ method is used.
+def new__str__(self):
+    """Change __str__ from pydantic ValidationError to include the file name if
+    present"""
+    if "ctx" in self.errors()[0] and "file" in self.errors()[0]["ctx"]:
+        return original__str__(self).replace(
+            "\n", f" in {self.errors()[0]['ctx']['file']}\n", 1
+        )
+    return original__str__(self)
+
+
+# Overwrite the original __str__ with new__str__
+pydantic.error_wrappers.ValidationError.__str__ = new__str__
+
+# Custom error class since we need to get the file information to ValidationError
+# See for details: https://pydantic-docs.helpmanual.io/usage/models/#custom-errors
+class DoubleNativeRegionError(PydanticValueError):
```

Comment:
Suggest to rename to `DuplicateRegionError` or `RegionConflictError` - Double isn't quite intuitive...